modules/dag-l0/src/main/scala/org/tessellation/dag/l0/domain/snapshot/programs/SnapshotFeeCalculator.scala

```diff
@@ -0,0 +1,26 @@
+package org.tessellation.dag.l0.domain.snapshot.programs
+
+import org.tessellation.node.shared.config.types.FeeCalculatorConfig
+import org.tessellation.security.signature.Signed
+import org.tessellation.statechannel.StateChannelSnapshotBinary
+
+trait SnapshotFeeCalculator {
+  def fee(binary: Signed[StateChannelSnapshotBinary]): Long
+}
+
+// NOTE: See https://www.notion.so/Network-Fees-on-the-Hypergraph-2024-e8cc69ae4472488a8b78d55ee9c04449
+object SnapshotFeeCalculator {
```

Comment:
IMO it would be great to instead of linking the document, express it using unit tests, so the calculator would be covered by tests and by reading the tests we would read the calculator specification and compare with the document statements